### PR TITLE
feat(filters): clean up all the old filter syntax

### PR DESCRIPTION
**Pull request by anehx**
BREAKING CHANGE:

We have for a long time had two syntaxes for filtering, one of which was
deprecated for over a year now and only kept for backwards
compatibility.  Finally, the old form has to go.

Old:
```graphql
query foo {
    allWorkItems(status: $status) {...}
}
```

New:
```graphql
query foo {
    allWorkItems(filter: [{status: $status}]) {...}
}
```

The new syntax also allows reusing the same filter type twice to narrow
down the result set even more. It also supports inverting the filters to
implement explicit exclusion. It is now time to get rid of the old
syntax.

The following changes are breaking:

- Removed deprecated `slug` filters in favor or `slugs` filters
- Removed top level filters
- Removed `orderByQuestionAnswerValue` filter in favor of
`documentAnswer` in the top level orderset
- Removed `orderBy` in filtersets in favor of top level `order` orderset
- Removed ordering by static meta attributes (configured with
`META_FIELDS` in favor of `meta` in the top level orderset

Resolves https://github.com/projectcaluma/caluma/issues/1489.
Closes https://github.com/projectcaluma/caluma/pull/1490

### CODE REVIEW

File: caluma/caluma_core/models.py [hunk 1]
```diff
@@ -109,18 +108,4 @@ class Meta:
...
-    """
-
-    def formfield(self, **kwargs):
-        # Postfixing newly created filter with Argument to avoid conflicts
-        # with query nodes
-        meta = self.model._meta
-        name = to_camel_case(f"{meta.object_name}_{self.name}_argument")
-
-        return super().formfield(label=name, **kwargs)
+    pass
```
Comment (winged) [inline caluma/caluma_core/models.py:111]: Do we even need this anymore?

File: caluma/caluma_form/filters.py [hunk 2]
```diff
@@ -486,7 +482,7 @@ class VisibleAnswerFilter(Filter):
     field_class = BooleanField
 
     def filter(self, qs, value):
-        if not value or not qs.exists():
+        if value in EMPTY_VALUES or not qs.exists():  # pragma: no cover
```
Comment (winged) [inline caluma/caluma_form/filters.py:485]: I was just gonna suggest removing the `qs.exists()` check. But given that it triggers loading a whole document structure and evaluating the visibility JEXL, maybe we should add a comment instead to explain why we're doing this here, while we're at it :-) Might be a bit unrelated but useful for the next person to look at this

**Review by winged — COMMENTED** [review]
Nice work on picking that up again, looking good so far!

**Review by winged — APPROVED** [review]

File: caluma/caluma_core/models.py (hunk 1 above)
Comment (anehx) [reply caluma/caluma_core/models.py]: Yes, for backwards compatiblity in the migrations. I added a comment and removed all implementation in favor of `models.CharField`

File: caluma/caluma_form/filters.py (hunk 2 above)
Comment (anehx) [reply caluma/caluma_form/filters.py:485]: Added a comment

**Comment by winged** [comment]
\m/

**Pull request merged**